src/components/space/SpaceScreen.js

```diff
@@ -106,6 +118,17 @@ class SpaceScreen extends Component {
       // eslint-disable-next-line react/no-did-update-set-state
       this.setState({ selected: -1 });
     }
+
+    if (space && !space.isEmpty()) {
+      const spaceId = space.get('id');
+      const recentIndex = recentSpaces.indexOf(spaceId);
+
+      // update recent spaces if space is not recent
+      // or space is not most recent space
```

Comment:
How are you referring to the most recent space here? Wouldn't it be `0`?

Reply:
Ah... I see. You're doing this in reverse.

Reply:
Yes it was easier to add spaces at the end of the List and remove the first element if the max number was reached.